pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
Actually, thinking about it again, most traffic is supposed to go from binary caches instead of fetching sources, so having https by default for source fetches sounds good. The actual checking of certs (``--insecure``) will probably have completely negligible overhead, given that you need to *build* the package after you fetch the sources.

BTW, adding that PGP scheme would be nice, maybe as opt-in via an impure env. variable for nixpkgs devs...